app.rb

```diff
@@ -38,17 +43,27 @@ def create_person
       puts 'Student has parent permission :'
       permission = gets.chomp
       student = Student.new(@people.length + 1, name, age, classroom: class_name, parent_permission: permission)
-      @people << student
+
+      @people << { 'id' => student.id, 'name' => student.name, 'age' => student.age, 'parent_permission' => student.parent_permission,
+                   'rentals' => student.rentals }
+
+      File.write('./data/people.json', JSON.generate(@people))
```

Comment:
- [ ]  I attempted to add a new `person` to test the changes suggested by the last reviewer on the error prompt when existing files are deleted, and i ran into a new error when imputing `parent permission`. Same issue occurred when i tried to list the  `books`. 
- [ ] The error prompt says the `parent_permission` is not defined. i suggest your review the class hierarchy and call methods to trace the error 

- [ ] Please try and add at least two values and test all functionality before submitting for another review 👍 

### Error shown:

![image](https://github.com/ridabensalem/School-library/assets/50990925/caa5e549-122b-4972-9e4b-c28d3f876364)